onnx/version_converter/IntraDomainConverter.h

```diff
@@ -0,0 +1,212 @@
+// Converter for ONNX models between different opset versions
+// in the same domain.
+
+#pragma once
+
+#include "onnx/version_converter/BaseConverter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+struct IntraDomainVersionConverter : BaseVersionConverter {
+  bool DEBUG = false;
+
+  IntraDomainVersionConverter() {
+    // TODO: Register adapters to the version converter
+  }
+
+  Adapter* adapter_lookup(Node* op,
+      const OpSetID& initial_version,
+      const OpSetID& target_version) {
+    std::string op_name = op->name();
+    std::string initial = initial_version.toString();
+    std::string target = target_version.toString();
+    // Find appropriate adapter in adapters map for provided initial and target versions
+    // TODO: Consider abstracting elements of this that are specific to
+    // DefaultConverter to separate methods here and maintain the procedure in Base Converter
+    if (adapters.find(op_name) != adapters.end()) {
+      // If we're adapting downwards, we just want to find the one downwards
+      // adapter implemented for initial_version. If we're adapting upwards, we
+      // want to actually use the SinceVersion value for the given op.
+      if (target_version.version < initial_version.version) {
+        // Downwards adapter
+        if (adapters[op_name].find(initial) != adapters[op_name].end()) {
+          // Either an upwards or a downwards adapter exists
+          // Check if downwards adapter exists (only one should)
```

Comment:
Is this always true, that there will only be exactly one up- or downwards adapter? If yes, we can encode it into the data structure and make it faster by having a nesting level less. You don't need a map if it always has only one entry.

Reply:
For this purpose, as we continue to evaluate the current version of adapter_lookup, it may make sense to move it back to the BaseConverter.